website/integrations: nextcloud: connect to existing user (#9155)
goauthentik/authentik: website/integrations/services/nextcloud/index.md

##### Custom profile scope

> If you do not need storage quota, group information or to manage already existing user in Nextcloud [skip to the next step](#provider-and-application).

Reviewer: ```suggestion
If you do not need storage quota, group information, or to manage an already existing user in Nextcloud [skip to the next step](#provider-and-application).
```

Author: I would put `already existing users` (plural instead)

If set to a value, for example `1 GB`, user(s) will have 1GB storage quota. If the attribute is not set, user(s) will have unlimited storage.
:::

:::note

> To connect to an already existing Nextcloud user, set the "nextcloud_user_id" property in the user's attributes. This must be set for individual users.

Reviewer: ```suggestion
To connect to an already existing Nextcloud user, set the "nextcloud_user_id" property in the user's attributes. This must be set for each individual user.
```

> The value must match the field `username` of the user on the Nextcloud instance. On Nextcloud, go to _Users_ -> check the username of the user you are trying to connect to (Under `Display name`).

Reviewer: ```suggestion
The value of `nextcloud_user_id` must match the field `username` of the user on the Nextcloud instance. On Nextcloud, go to _Users_ to see the username of the user you are trying to connect to (Under `Display name`).
```

Reviewer: @Fymyte please double-check my edits to make sure I did not change the meaning, or how the UI actually appears... I wasn't sure what the arrow (->) was for?

Author: Thanks, the arrow was here because I thought there would be more sub-menus 😅 it had no meaning

    :::
-   Scope: `email`, `profile` (you can safely omit `openid` if you prefer)
-   Attribute mappings:

>     -   User ID mapping: sub (or `user_id` if you need to connect to already existing Nextcloud user)

Reviewer: ```suggestion
    -   User ID mapping: sub (or `user_id` if you need to connect to an already existing Nextcloud user)
```

    -   Display name mapping: name
    -   Email mapping: email
    -   Quota mapping: quota (leave empty if you have skipped the [custom profile scope](#custom-profile-scope) section)
    -   Groups mapping: groups (leave empty if you have skipped the [custom profile scope](#custom-profile-scope) section)
        :::tip
        You need to enable the "Use group provisioning" checkmark to be able to write to this field
        :::

> -   Use unique user ID: If you only have one provider you can uncheck this if you prefer (This will affect your Federated Cloud ID, under _Personal settings_ -> _Sharing_ -> _Federated Cloud_).

Reviewer: ```suggestion
-   Use unique user ID: If you only have one provider you can deselect this if you prefer (This will affect your Federated Cloud ID, under _Personal settings_ -> _Sharing_ -> _Federated Cloud_).
```

Reviewer: @Fymyte is it worth explaining exactly HOW deselecting this field will affect the ID?

Author: I am not 100% sure. If I am not mistaken, if you tick the box the user id will be a random UUID generated by nextcloud, so you will have something like `0101830273902827722@nextcloud.instance`.
If the box is not ticked, the id will be the one of the `user_id` set by Authentik, so Authentik's user uuid, or `nextcloud_user_id` if set.
This would allow, for example, to get a federated cloud id `Fymyte@nextcloud.instance` if you set the `nextcloud_user_id`.

Author: I tried to explain it better. Can you re-check ?

Reviewer: Hi @Fymyte sorry I was out on vacation. Thanks for that additional info on line 150.

Author: NP thanks for the feedbacks
